recipes/ftxui/all/conanfile.py

```diff
@@ -0,0 +1,73 @@
+import os
+
+import conan.tools.files
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class FTXUIConan(ConanFile):
+    name = "ftxui"
+    description = "💻 C++ Functional Terminal User Interface."
+    license = "MIT"
+    topics = ("ncurses", "terminal", "screen", "tui")
+    homepage = "https://github.com/ArthurSonzogni/FTXUI"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _validate_compiler_settings(self):
+        compiler = self.settings.compiler
+        version = tools.Version(self.settings.compiler.version)
+        if compiler == 'gcc' and version < '8':
+            raise ConanInvalidConfiguration("gcc 8 required")
+        if compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+
+    def validate(self):
+        self._validate_compiler_settings()
+        if self.settings.compiler in ["Visual Studio", "msvc"]:
+            raise ConanInvalidConfiguration("Visual Studio unsupported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+        libs = ['ftxui-dom', 'ftxui-screen', 'ftxui-component']
+        ext = '.a'
+
+        for lib in libs:
+            src = os.path.join(self.package_folder, "lib", f"{lib}{ext}")
+            dst = os.path.join(self.package_folder, "lib", f"lib{lib}{ext}")
+            conan.tools.files.rename(self, src, dst)
```

Comment:
FYI, it is [this line](https://github.com/ArthurSonzogni/FTXUI/blob/5da7b8a59ae812002c52bc0459947e90ec744eb4/cmake/ftxui_set_options.cmake#L2) which removes the "lib" prefix

Reply:
Reading the commit message, it seems this was to support the vcpkg package manager, and cause bug for the conan package manager.

Anyhow, I think a rename like I did is better than patching upstream.

Reply:
let's see what they say about https://github.com/ArthurSonzogni/FTXUI/pull/340
It would avoid renaming the libraries manually, and guess their extension

Reply:
But do you want us to wait to a new release with the fix before giving any recipe?

Reply:
No, We could either rename manually(as currently) or backport the patch if it is validated upstream

Reply:
IMHO it's better to keep the rename, as it's easier to maintain than a patch. For the new release we can either condition that in a version, or, move it to a specific script.

Reply:
ideally, we don't want to maintain patches - it's an additional burden to rebase them for new releases.
if patches could be avoided, let's try to avoid them. it should be the last resort if other ways didn't work.

Reply:
Well, it will be super easy, barely an inconvenience, because the patch has been merged upstream. So we won't have to maintain it. Also, the patch is a single line change, so we could use tools.replace_in_file

Reply:
Let me know what you decide and I'll use this approach.

What else is left?

Reply:
I think it's fine like it's now. only need to have green CI.